File: official/docs/java/current/shipping-insurance/insure.java
package shipments;

import com.easypost.exception.EasyPostException;
import com.easypost.model.Shipment;
import com.easypost.service.EasyPostClient;

import java.util.HashMap;

public class Insure {
    public static void main(String[] args) throws EasyPostException {
        EasyPostClient client = new EasyPostClient(System.getenv("EASYPOST_API_KEY"));

        HashMap<String, Object> insureMap = new HashMap<String, Object>();
        insureMap.put("amount", 100);

        Shipment shipment = client.shipment.insure("shp_...", insureMap);

        System.out.println(shipment);
    }
}

File: official/docs/php/current/shipping-insurance/insure.php
<?php

$client = new \EasyPost\EasyPostClient(getenv('EASYPOST_API_KEY'));

$shipment = $client->shipment->retrieve('shp_...');

$shipmentWithInsurance = $client->$shipment->insure(
    $shipment->id,
    ['amount' => 100]
);

echo $shipmentWithInsurance;
